Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Link/Link.cpp

```diff
@@ -282,6 +284,35 @@ namespace AzToolsFramework
             }
         }
 
+        PrefabOverridePrefixTree Link::GenerateOverrideSubTreeForEntity(
+            const PrefabDomValue& subpathPatches, const AZStd::string& pathToEntity)
+        {
+            PrefabOverridePrefixTree prefixTree;
+            if (subpathPatches.IsArray())
+            {
+                rapidjson::GenericArray patchesArray = subpathPatches.GetArray();
+                for (rapidjson::SizeType i = 0; i < patchesArray.Size(); i++)
+                {
+                    PrefabDom patchEntry;
+                    patchEntry.CopyFrom(patchesArray[i], patchEntry.GetAllocator());
+
+                    auto pathIter = patchEntry.FindMember("path");
+                    if (pathIter != patchEntry.MemberEnd())
+                    {
+                        AZStd::string subPath = pathIter->value.GetString(); // eg: '/Components/Component_[123]/Child Sort Order/0'
+                        AZStd::string fullPath = pathToEntity + subPath;
+                        pathIter->value.SetString(fullPath.c_str(), static_cast<rapidjson::SizeType>(fullPath.length()),
+                            patchEntry.GetAllocator());
```

Comment:
@srikappa-amzn I think the purpose I exposed the subtree API is I want to manage **subtree states** in undo node. The moment it generates the subtree it needs to finalize the JSON patch stored in the subtree; in other words, the indices stored in the patches need to be set up correctly. To be able to use the right indices, I changed the function from util to non-static member to understand the internal details of Link class.

So if I change to `AddOverrides(patchArray)`, it means it will manage **patch arrays** in undo node rather than subtree states.

I am still investigating how to improve this along with the indexing concern mentioned below.

Reply:
I didn't mean to change the implementation detail. I just think that a public function in the link class is the wrong place for this function to be in. If you look at this function, all it does is increment m_patchIndexCounter. If that's the case, we might as well create a function called SetPatchIndexCounter() and call just that and move everything else to a static member function in PrefabUndoEntityOverrides.cpp